auth/store.go

```diff
@@ -706,6 +706,10 @@ func (as *authStore) isOpPermitted(userName string, revision uint64, key, rangeE
 		return nil
 	}
 
+	if userName == "" {
+		return ErrUserEmpty
```

Comment:
as.revision should never be zero if it contains a valid info, right? should we also fix line713? or probably never call these permitted func when rev is 0?